packages/@tinacms/cli/src/next/codegen/index.ts

```diff
@@ -11,25 +14,61 @@ export class Codegen {
   schema: GraphQLSchema
   queryDoc: string
   fragDoc: string
+  graphqlSchemaDoc: {
+    kind: 'Document'
+    definitions: TypeDefinitionNode[]
+  }
+  tinaSchema: TinaSchema
+  lookup: any
 
   constructor({
     configManager,
     port,
-    schema,
     queryDoc,
     fragDoc,
+    graphqlSchemaDoc,
+    tinaSchema,
+    lookup,
   }: {
     configManager: ConfigManager
     port?: number
-    schema: GraphQLSchema
     queryDoc: string
     fragDoc: string
+    graphqlSchemaDoc: {
+      kind: 'Document'
+      definitions: TypeDefinitionNode[]
+    }
+    tinaSchema: TinaSchema
+    lookup: any
   }) {
+    this.graphqlSchemaDoc = graphqlSchemaDoc
     this.configManager = configManager
     this.port = port
-    this.schema = schema
+    this.schema = buildASTSchema(graphqlSchemaDoc)
+    this.tinaSchema = tinaSchema
     this.queryDoc = queryDoc
     this.fragDoc = fragDoc
+    this.lookup = lookup
+  }
+
+  async writeConfigFile(fileName: string, data: string) {
+    const tinaFolderPath = await this.configManager.getTinaFolderPath(
+      this.configManager.rootPath
+    )
+    const filePath = path.join(tinaFolderPath, GENERATED_FOLDER, fileName)
```

Comment:
```suggestion
    const filePath = path.join(this.configManager.generatedFolderPath, fileName)
```

Reply:
@jeffsee55 what about the one on line 68 below it?

Reply:
Yes sorry same thing in both places.

Reply:
@jeffsee55 Do we have a generatedFolder path for both the rootPath and the contentRootPath? id like to keep it consistent and leave it as is if we don't

Reply:
Oh sorry, I misunderstood that. No we don't have that, but would be handy to have as part of the config manager.